### ISSUE TITLE
Use a linter preset such as airbnb to standardise code

### ISSUE THREAD

**Issue by andrewtobin** [CodeSeven/toastr#465]
Hey guys,

This is really low priority and probably "poking bears", but I was wondering about introducing a preset for jscs/jshint to standardise code style being committed.

Mainly this is because I know how terrible some of my own code can be and I like having a preset that encourages me to standardise to a slightly more opinioned standard before acceptance.

http://jscs.info/overview
https://github.com/airbnb/javascript

**Comment by TimFerrell** [comment]
I am in support of this.

**Comment by TimFerrell** [comment]
I haven't used airbnb before. Have used jslint and jshint before, and read that ESLint has taken over as king. Mind jotting down a quick comparison of the two?

**Comment by andrewtobin** [comment]
Sure.  It's not actually a linter, it's just a set of rules for javascript applied in jscs or jshint, which are both in the project.  Basically, it's an opinionated ruleset that is a bit more stringent than "passes ES2015" like the current set, and includes the rules in the wiki linked.

Things like:

- Warn to replace var with let/const
- Prefer object destructuring when accessing multiple properties of an object.
- Prefer single quotes for strings.
- Don't mutate or reassign parameters.
- Prefer Arrow Functions
- etc

Basically, it means you're writing slightly stricter js, but means it's more consistent.

I'll apply the ruleset and alter the code to pass and make a pull request and you can see what I mean, and adopt or reject it :)

**Comment by andrewtobin** [comment]
Just to add, if you take a look at the header on [jscs.info](http://jscs.info/) they've merged projects with ESLint as it is anyway :)

**Comment by TimFerrell** [comment]
Closed via recent PR.
